lib/queues.js

```diff
@@ -260,13 +270,62 @@ class Queue {
   runOnce() {
     assert(!this._processing, 'Cannot call once while continuously processing jobs');
     const hasHandlers = (this._handlers.size > 0);
+
     if (hasHandlers) {
 
       debug('Waiting for jobs on queue %s', this.name);
-      return this._getClientPromise.then(client => this._reserveAndProcess(client));
+
+      if (process.env.NODE_ENV === 'test' && this._delayedJobs.size > 0) {
```

Comment:
- `runOnce()` _is_ for testing, so maybe we don't need to check `NODE_ENV` here.
- Anyway, we are checking `NODE_ENV` to populate `_delayedJobs`, so we can trust that if there's anything in there, we want to check for it here.
- `_delayedJobs` is always a collection (empty or not), so maybe we can avoid the `if`?